pkg/controller/mcmhub/mcmhub_controller.go

```diff
@@ -177,13 +179,120 @@ type ReconcileSubscription struct {
 	eventRecorder *utils.EventRecorder
 }
 
+// CreateSubscriptionAdminRBAC checks existence of subscription-admin clusterrole and clusterrolebinding
+// and creates them if not found
+func (r *ReconcileSubscription) CreateSubscriptionAdminRBAC() error {
+	// Create subscription admin ClusteRole
+	clusterRole := subAdminClusterRole()
+	foundClusterRole := &rbacv1.ClusterRole{}
+
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: clusterRole.Name}, foundClusterRole); err != nil {
+		if k8serrors.IsNotFound(err) {
+			klog.Info("Creating ClusterRole ", clusterRole.Name)
+			err = r.Create(context.TODO(), clusterRole)
+
+			if err != nil {
+				klog.Error("error:", err)
+				return err
+			}
+		} else {
+			klog.Error("error:", err)
+			return err
+		}
+	}
+
+	// Create subscription admin ClusteRoleBinding
+	clusterRoleBinding := subAdminClusterRoleBinding()
+	foundClusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: clusterRoleBinding.Name}, foundClusterRoleBinding); err != nil {
+		if k8serrors.IsNotFound(err) {
+			klog.Info("Creating ClusterRoleBiding ", clusterRoleBinding.Name)
+			err = r.Create(context.TODO(), clusterRoleBinding)
+
+			if err != nil {
+				klog.Error("error:", err)
+				return err
+			}
+		} else {
+			klog.Error("error:", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func subAdminClusterRole() *rbacv1.ClusterRole {
```

Comment:
I added some other api groups and removed the clusterregistry api group base on the new spoke cluster api refactoring. Please update the role 
1. add the following api groups:
```
        - apiGroups:
          - "admissionregistration.k8s.io"
          resources:
          - "*"
          verbs:
          - "*"
        - apiGroups:
          - "cluster.open-cluster-management.io"
          resources:
          - "*"
          verbs:
          - "*"
        - apiGroups:
          - "coordination.k8s.io"
          resources:
          - "leases"
          verbs:
          - "*"
```
2. remove clusterregistry api group